### PR TITLE
(#1453) Migrate test to JUnit 5

### PR DESCRIPTION
**Pull request by baudoliver7**
For #1453. Migrate tests to JUnit 5 for test packages :

- `org.cactoos`
- `org.cactoos.bytes`
- `org.cactoos.collection`
- `org.cactoos.experimental`

### CODE REVIEW

**Comment by baudoliver7** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/cactoos/issues/1580#issuecomment-817194935))

@baudoliver7 This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1580](https://github.com/yegor256/cactoos/issues/1580) is [in scope](http://www.zerocracy.com/policy.html#14) for 5days
  * The role is `REV`
  * The job is assigned to @andreoss[/z](https://www.0crat.com/u/andreoss) for [5days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @baudoliver7[/z](https://www.0crat.com/u/baudoliver7): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Ayegor256%2Fcactoos%231580%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by baudoliver7** [comment]
@andreoss ping

File: src/test/java/org/cactoos/bytes/HexOfTest.java [hunk 1]
```diff
@@ -66,13 +68,19 @@ public void validHex() throws Exception {
...
+        ).affirm();
     }
 
-    @Test(expected = IOException.class)
     public void invalidHex() throws Exception {
-        new HexOf(new TextOf("ABG!")).asBytes();
+        new Assertion<>(
+            "Must invalid hex",
+            () -> new HexOf(new TextOf("ABG!")).asBytes(),
+            new Throws<>(IOException.class)
```
Comment (andreoss) [inline src/test/java/org/cactoos/bytes/HexOfTest.java:83]: @baudoliver7 It would be better to also match message of exception

**Review by andreoss — CHANGES_REQUESTED** [review]

File: src/test/java/org/cactoos/bytes/HexOfTest.java (hunk 1 above)
Comment (baudoliver7) [reply src/test/java/org/cactoos/bytes/HexOfTest.java]: @andreoss Done.

**Comment by baudoliver7** [comment]
@andreoss Please, see my changes.

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (14 days), architects (@victornoel) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**